File: event-schemas/src/entities/MapContext.ts


class MapContext {
  private _id?: string;
  private _name?: string;
  private _level?: number;
  private _position?: number;
  private _additionalProperties?: Map<String, object | string | number | Array<unknown> | boolean | null>;

  constructor(input: {
    id?: string,
    name?: string,
    level?: number,
    position?: number,
  }) {
    this._id = input.id;
    this._name = input.name;
    this._level = input.level;
    this._position = input.position;
  }

  get id(): string | undefined { return this._id; }
  set id(id: string | undefined) { this._id = id; }

  get name(): string | undefined { return this._name; }
  set name(name: string | undefined) { this._name = name; }

  get level(): number | undefined { return this._level; }
  set level(level: number | undefined) { this._level = level; }

  get position(): number | undefined { return this._position; }
  set position(position: number | undefined) { this._position = position; }

  get additionalProperties(): Map<String, object | string | number | Array<unknown> | boolean | null> | undefined { return this._additionalProperties; }
  set additionalProperties(additionalProperties: Map<String, object | string | number | Array<unknown> | boolean | null> | undefined) { this._additionalProperties = additionalProperties; }
}
export default MapContext;

File: event-schemas/data/csharp/MapContext.cs
namespace entities
{
  using System.Collections.Generic;

  public class MapContext
  {
    private string id;
    private string name;
    private double? level;
    private double? position;
    private Dictionary<string, dynamic> additionalProperties;

    public string Id 
    {
      get { return id; }
      set { id = value; }
    }

    public string Name 
    {
      get { return name; }
      set { name = value; }
    }

    public double? Level 
    {
      get { return level; }
      set { level = value; }
    }

    public double? Position 
    {
      get { return position; }
      set { position = value; }
    }

    public Dictionary<string, dynamic> AdditionalProperties 
    {
      get { return additionalProperties; }
      set { additionalProperties = value; }
    }
  }
}